File: Presentation Layer (frontend)/Reservation_angular/src/app/service/customer.service.ts

import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { Login } from 'src/data/login';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  headerDict = {
    'Content-Type': 'application/json',
    'Accept': 'application/json'
  }
  requestOptions = {
    headers: new HttpHeaders(this.headerDict)
  }
  constructor(public http: HttpClient, private route: Router) { }
  //dto
  GetAllCustomers() {
    return this.http.get(environment.apiUrl + "customer/GetAllCustomers", this.requestOptions)
  }
  GetCustomerByName(name: string) {
    return this.http.get(environment.apiUrl + "Customer/GetCustomerByName?name=" + name, this.requestOptions)
  }
  FilterCustomerByName(name: string) {
    return this.http.get(environment.apiUrl + "Customer/FilterCustomerByName?name=" + name, this.requestOptions)
  }
  GetCustomerById(id: number) {
    return this.http.get(environment.apiUrl + "Customer/GetCustomerById?id=" + id, this.requestOptions)

  }
  GetRegisterByRegisterId(regid:number){

    return this.http.get<any[]>(environment.apiUrl +"Register/GetRegisterByRegisterId?regId=" + regid, this.requestOptions)


  }

  UpdateRegister(id :number,name :string ,phone:string,password:string ){
    return this.http.put(environment.apiUrl+"Register/UpdateRegister?id="+id+"&name="+name+"&phone="+phone+"&password="+password,this.requestOptions);
  }



}

File: Business Logic Layer(Backend)/Reservation/Reservation.Infra/Service/CustomerService.cs
using Reservation.Core.Data;
using Reservation.Core.DTO;
using Reservation.Core.Repository;
using Reservation.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;

namespace Reservation.Infra.Service
{
    public class CustomerService:ICustomerService
    {
        private readonly ICustomerRepository _customerRepository;

        public CustomerService(ICustomerRepository customerRepository)
        {
            _customerRepository = customerRepository;
        }

        public List<CustomerDTO> GetAllCustomers()
        {
            return _customerRepository.GetAllCustomers();
        }
        public List<CustomerDTO> GetCustomerByName(string name)
        {
            return _customerRepository.GetCustomerByName(name);
        }
        public List<CustomerDTO> FilterCustomerByName(string name)
        {
            return _customerRepository.FilterCustomerByName(name);
        }
        public List<CustomerDTO> GetCustomerById(int id)
        {
            return _customerRepository.GetCustomerById(id);
        }
    }
}
